vector/src/test/scala/spec/geotrellis/vector/ExtentSpec.scala

```diff
@@ -18,9 +18,12 @@ package geotrellis.vector
 
 import geotrellis.vector.io._
 import geotrellis.vector.io.json.JsonFeatureCollection
+
 import org.scalatest.FunSpec
 import org.scalatest.Matchers
 import spray.json.DefaultJsonProtocol._
+// `import cats.implicits._` causes ambiguous implicits on `===`:  https://github.com/typelevel/cats/issues/219
```

Comment:
A good reason to use `import cats.syntax.semigroup._` everywhere instead of `implicits`.

Reply:
@pomadchin @moradology Seems like `import cats.syntax.semigroup._` is going to be required import in lots of places. It looks like we're replacing the geotrellis imports soup with cats import soup.

When faced with this problem before I've chosen to leave the `combine` methods on objects that currently have them and add the `Semigroup` instance that uses that. That at least has the potential of using the method `.combine` when you know the explicit type without import and use the `Semigroup[K].combine` when you're writing abstract code. What do you guys think of that tactic ?

Reply:
@echeipesh I don't have a strong opinion about that, I had some thoughts about it but decided to await until the PR would be +- ready. I also wanted to discuss in general do we need `Semigroups` in places where it is not obvious. Since we don't have a code that would use it, does it make any sense to add it? Mb it is a part of a user application in the case we don't use them. Tbh I would like not to include the dead code.

Reply:
@echeipesh eh, I said yesterday Nathan to push changes into a separate commit ): but now it's squashed... The story is that he implemented these things you're asking about in a first commit, and in the second he covered #2925... you don't see the whole story since it is a force push ):

Reply:
I ran into this for polygonal summary API. It was pretty annoying to include the `cats.syntax.semigroup._` everywhere in my concrete implementations, and its sort of an open secret that some people (read: including me until @moradology told me about it) might not know about which increases friction for GT users when there's lots of other stuff to learn. 

In my case, there wasn't any external API for polygonal summary that exposed the requirement to include the cats import soup so this concern didn't come up there, but if one of our design goals is to reduce imports and increase clarity, then taking @echeipesh 's suggestion to keep explicit combine methods that internally use cats rather than pushing that out to the user seems like a real good idea. Isn't that what abstraction is for anyways?